app/web/src/store/change_sets.store.ts

```diff
@@ -603,6 +604,21 @@ export function useChangeSetsStore() {
                 const route = useRouterStore().currentRoute;
 
                 if (route?.name) {
+                  // am i on a view, and about to move to head
+                  // where that view does not yet exist
+                  if ("viewId" in route.params) {
```

Comment:
Should this be generic, for switching between any change set? Or does it only matter when going to head

Reply:
good point!

Reply:
this spot is only about moving to head after apply... i will add the changeset dropdown